nixos/modules/services/web-servers/caddy/default.nix

```diff
@@ -171,34 +171,24 @@ in
   };
 
   config = mkIf cfg.enable {
+    systemd.packages = [ cfg.package ];
```

Comment:
Sorry to ask a noob question, but I'm not very familiar with this systemd.packages option.
From what I understand it sets up systemd files from the given packages https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/system/boot/systemd.nix#L955
It looks like in the code, there is a `type` parameter though that is needed. Does it work by just putting a file inside $package/lib/systemd ?

My other question pertains to what happens to the rest of the unit defined here ? Like the ExecStart and ExecReload are in conflict with the ones supplied by the upstream service unit right ? How do those get merged ?

Reply:
Good question. I haven't ever really gone through the implementation in detail. `caddy-api.service` won't be activated, though, because we don't add a `wantedBy` declaration to it in the module. See https://github.com/NixOS/nixpkgs/issues/81138 for details.

With `systemd` you can reset prior values of `ExecStart` by providing an empty value in your override, hence the `[ "<empty>" "<actual command>" ]` - so what I have in `ExecStart` and `ExecReload` completely tramples the upstream values.

Reply:
Ok, let me test that tomorrow then. (just to make sure it works fine).

Should we add a comment about overriding a value by adding an empty string ? (If you feel this is standard enough, then feel free not to, just thought it might be easier for other people to understand in case it's one of the more exotic part of nixos).

Reply:
Thank you very much! I have not migrated to `caddy` yet (but am keen to do so on one of my servers soon) so testing is required and appreciated.

This isn't a NixOS feature really... moreso a `systemd` feature. While it is a standard feature and is used often enough in NixOS clearly by you asking this it justifies adding a comment. Something quick and simple should suffice:

```
# https://www.freedesktop.org/software/systemd/man/systemd.service.html#ExecStart=
# If the empty string is assigned to this option, the list of commands to start is reset, prior assignments of this option will have no effect.
```

Sound good?

Reply:
Sounds amazing!